src/odemis/driver/orsay.py

```diff
@@ -1356,23 +1242,12 @@ def _setTemperatureTarget(self, goal):
 
     def _setTemperatureRegulation(self, goal):
         """
-        goal (int): mode to set the temperature regulation to. Must be 0, 1 or 2
+        goal (boolean): mode to set the temperature regulation to. True is on, False is off.
 
-        Turns temperature regulation off (if goal = 0), on (if goal = 1) or on in accelerated mode (if goal = 2)
+        Turns temperature regulation off (if goal = False) or on (if goal = True)
         """
-        reg = False
-        rush = False
-        if goal == 2:
-            logging.debug("Setting temperature regulation to accelerated mode.")
-            rush = True
-            reg = True
-        elif goal == 1:
-            logging.debug("Turning on temperature regulation.")
-            reg = True
-        else:
-            logging.debug("Turning off temperature regulation.")
-        self._gis.RegulationOn.Target = reg
-        self._gis.RegulationRushOn.Target = rush
+        logging.debug("Turning temperature regulation %s." % "on" if goal else "off")
+        self._gis.RegulationOn.Target = goal
```

Comment:
The setters *must* always return the new value that the VA will hold. So, here, you should return "goal".